backend/handler/auth/base_handler.py

```diff
@@ -164,3 +156,55 @@ async def get_current_active_user_from_bearer_token(self, token: str):
             )
 
         return user, payload.claims
+
+
+class OpenIDHandler:
+    def __init__(self) -> None:
+        if not OIDC_ENABLED:
+            return
+
+        # Fetch the public key from the OIDC server
+        # JWKS (JSON Web Key Sets) response is a JSON object with a keys array
+        jwks_url = f"{OIDC_SERVER_APPLICATION_URL}/jwks/"
+        with httpx.Client() as httpx_client:
+            try:
+                response = httpx_client.get(jwks_url, timeout=120)
+                key = response.json()["keys"][0]
+                self.rsa_key = RSAKey.import_key(key)
+            except httpx.HTTPStatusError as exc:
+                raise HTTPException(
+                    status_code=exc.response.status_code,
+                    detail=exc.response.text,
+                ) from exc
```

Comment:
I think it would be better to move this logic to a method, to avoid us triggering an HTTP request at import time (even if the `get_current_active_user_from_openid_token` method is never called, like in the `watcher.py` process).

```python
    async def get_rsa_key(self) -> RSAKey:
        if not getattr(self, "_rsa_key", None):
            try:
                response = await httpx_client.get(jwks_url, timeout=120)
                key = response.json()["keys"][0]
                self._rsa_key = RSAKey.import_key(key)
            except httpx.HTTPStatusError as exc:
                raise HTTPException(
                    status_code=exc.response.status_code,
                    detail=exc.response.text,
                ) from exc
        return self._rsa_key
```